x-pack/plugins/cloud_security_posture/server/routes/status/status.ts

```diff
@@ -60,31 +67,46 @@ const getHealthyAgents = async (
   );
 };
 
-const calculateCspStatusCode = (
+export const calculateCspStatusCode = (
+  postureType: PostureTypes,
   indicesStatus: {
     findingsLatest: IndexStatus;
     findings: IndexStatus;
     score: IndexStatus;
   },
   installedCspPackagePolicies: number,
   healthyAgents: number,
-  timeSinceInstallationInMinutes: number
+  timeSinceInstallationInMinutes: number,
+  installedPolicyTemplates: string[]
 ): CspStatusCode => {
   // We check privileges only for the relevant indices for our pages to appear
+  const postureTypeCheck =
+    postureType === CSPM_POLICY_TEMPLATE ? CSPM_POLICY_TEMPLATE : KSPM_POLICY_TEMPLATE;
   if (indicesStatus.findingsLatest === 'unprivileged' || indicesStatus.score === 'unprivileged')
     return 'unprivileged';
-  if (indicesStatus.findingsLatest === 'not-empty') return 'indexed';
-  if (installedCspPackagePolicies === 0) return 'not-installed';
+  if (!installedPolicyTemplates.includes(postureTypeCheck)) return 'not-installed';
   if (healthyAgents === 0) return 'not-deployed';
-  if (timeSinceInstallationInMinutes <= INDEX_TIMEOUT_IN_MINUTES) return 'indexing';
-  if (timeSinceInstallationInMinutes > INDEX_TIMEOUT_IN_MINUTES) return 'index-timeout';
+  if (
+    indicesStatus.findingsLatest === 'empty' &&
+    indicesStatus.findings === 'empty' &&
+    timeSinceInstallationInMinutes < INDEX_TIMEOUT_IN_MINUTES
+  )
+    return 'waiting_for_results';
+  if (
+    indicesStatus.findingsLatest === 'empty' &&
+    indicesStatus.findings === 'empty' &&
+    timeSinceInstallationInMinutes > INDEX_TIMEOUT_IN_MINUTES
+  )
+    return 'index-timeout';
+  if (indicesStatus.findingsLatest === 'empty') return 'indexing';
+  if (indicesStatus.findings === 'not-empty') return 'indexed';
 
```

Comment:
the `indexed` status is only when the latest findings index contains data and here you change it to check the findings index.

Reply:
latest findings index has a data retention for a couple hours right ? Based on what me and Kfir discussed back then, Kfir mention that as long as findings exist (even from back then) status = indexed